modules/core/src/v2/coins/avaxc.ts

```diff
@@ -35,24 +352,330 @@ export class AvaxC extends AbstractEthLikeCoin {
+    if (!recipients || !Array.isArray(recipients)) {
+      throw new Error('expecting array of recipients');
+    }
+
+    // Right now we only support 1 recipient
+    if (recipients.length !== 1) {
+      throw new Error('must send to exactly 1 recipient');
+    }
+    const recipientAddress = recipients[0].address;
+    const recipientAmount = recipients[0].amount;
+    const feeEstimateParams = {
+      recipient: recipientAddress,
+      amount: recipientAmount,
+      hop: true,
+    };
+    const feeEstimate: FeeEstimate = await this.feeEstimate(feeEstimateParams);
+
+    const gasLimit = feeEstimate.gasLimitEstimate;
+    const gasPrice = Math.round(feeEstimate.feeEstimate / gasLimit);
+    const gasPriceMax = gasPrice * 5;
+    // Payment id a random number so its different for every tx
+    const paymentId = Math.floor(Math.random() * 10000000000).toString();
+    const hopDigest: Buffer = AvaxC.getHopDigest([
+      recipientAddress,
+      recipientAmount,
+      gasPriceMax.toString(),
+      gasLimit.toString(),
+      paymentId,
+    ]);
+
+    const userReqSig = optionalDeps.ethUtil.addHexPrefix(
+      Buffer.from(secp256k1.ecdsaSign(hopDigest, userPrvBuffer).signature).toString('hex')
+    );
+
     return {
-      halfSigned: {
-        txHex: transaction.toBroadcastFormat(),
-        recipients: recipients,
-        expiration: params.txPrebuild.expireTime,
+      hopParams: {
+        gasPriceMax,
+        userReqSig,
+        paymentId,
       },
+      gasLimit,
     };
   }
+
+  /**
+   * Fetch fee estimate information from the server
+   * @param {Object} params The params passed into the function
+   * @param {Boolean} [params.hop] True if we should estimate fee for a hop transaction
+   * @param {String} [params.recipient] The recipient of the transaction to estimate a send to
+   * @param {String} [params.data] The ETH tx data to estimate a send for
+   * @returns {Object} The fee info returned from the server
+   */
+  async feeEstimate(params: FeeEstimateOptions): Promise<FeeEstimate> {
+    const query: FeeEstimateOptions = {};
+    if (params && params.hop) {
+      query.hop = params.hop;
+    }
+    if (params && params.recipient) {
+      query.recipient = params.recipient;
+    }
+    if (params && params.data) {
+      query.data = params.data;
+    }
+    if (params && params.amount) {
+      query.amount = params.amount;
+    }
+
+    return await this.bitgo.get(this.url('/tx/fee')).query(query).result();
+  }
+
+  /**
+   * Gets the hop digest for the user to sign. This is validated in the HSM to prove that the user requested this tx
+   * @param paramsArr The parameters to hash together for the digest
+   */
+  private static getHopDigest(paramsArr: string[]): Buffer {
+    const hash = new Keccak('keccak256');
+    hash.update([AvaxC.hopTransactionSalt, ...paramsArr].join('$'));
+    return hash.digest();
+  }
+
+  isWalletAddress(params: VerifyAddressOptions): boolean {
+    // TODO: Fix this later
```

Comment:
what's the plan to implement this?

Reply:
https://bitgoinc.atlassian.net/browse/BG-43225 - https://bitgoinc.atlassian.net/browse/BG-43269 - looks like they gonna implement this next week.